src/renderers/terminal.rs

```diff
@@ -0,0 +1,136 @@
+use crate::{HighlightedCode, RenderOptions, themes::compiled::ThemeType};
+
+/// Terminal renderer via ANSI escape codes. Requires a terminal that supports truecolor
+#[derive(Default, Copy, Clone, PartialEq, Eq)]
+pub struct TerminalRenderer {
+    /// The theme type to use if [`ThemeVariant::Dual`](crate::ThemeVariant::Dual) is provided, since terminals don't allow light or dark theme
+    pub theme_type: ThemeType,
```

Comment:
It should be an Option so users don't need to specify it if they use ThemeVariant::Signle